### PR TITLE
Add drone support for loki operator/v* tags

### PR DESCRIPTION
**Pull request by periklis**
**What this PR does / why we need it**:
Adds support to handle upcoming `operator/v*` tags for Loki Operator releases to start using tags instead of git refs. Starting with: 
- grafana/loki/pull/10050

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

File: .drone/drone.yml [hunk 1]
```diff
@@ -1209,16 +1218,32 @@ steps:
...
+depends_on:
+- lokioperator-amd64
+- lokioperator-arm64
+- lokioperator-arm
+kind: pipeline
+name: manifest-operator
+steps:
 - depends_on:
   - clone
-  - manifest-loki-canary
```
Comment (dannykopping) [inline .drone/drone.yml:1214]: This doesn't look right?

Comment (periklis) [reply .drone/drone.yml:1214]: It's a entire new pipeline, that's ok

Comment (dannykopping) [reply .drone/drone.yml:1214]: Oh I see; this is a diff issue

**Review by dannykopping — APPROVED** [review]
LGTM

**Pull request merged**